lib/url-sanitizer.js

```diff
@@ -61,8 +61,8 @@ function safeDecodeURI (path) {
       }
     // Some systems do not follow RFC and separate the path and query
     // string with a `;` character (code 59), e.g. `/foo;jsessionid=123456`.
-    // Thus, we need to split on `;` as well as `?` and `#`.
-    } else if (charCode === 63 || charCode === 59 || charCode === 35) {
+    // Thus, we need to split on `;` as well as `?` and `#` if the useSemicolonDelimiter option is enabled.
+    } else if (charCode === 63 || charCode === 35 || (useSemicolonDelimiter && charCode === 59)) {
       querystring = path.slice(i + 1)
```

Comment:
If there is any perf difference (I doubt), you can move the `useSemicolonDelimiter`additional check under the `charCode` checking `if`. That would remove it from a hot path.

Reply:
Yes, given that the `;` symbol is rare, that's correct. I fixed